contracts/Hangman.sol

```diff
@@ -51,27 +51,34 @@ contract Hangman is Ownable {
         require(currentMisses < maxAllowedMisses, "no more guesses available");
         require(playerInput < 2**(solution.length), "solution is found"); // becareful of overflow
 
-        emit TurnTaken();
-
         //go through and check if the character has already been guessed
         for (uint i = 0; i < usedCharacters.length; i++) {
-            require(usedCharacters[i] != _character, "character has aleady been guessed");
```

Comment:
removal of this results in accepting already use characters again. This will consume gas instead of preventing the user from guessing this char.